### PR TITLE
chore: Bump OpenDAL to v0.37

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR will bump OpenDAL to v0.37 for bug fixes.

Changelogs: https://docs.rs/opendal/0.37.0/opendal/docs/changelog/index.html

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### Types of user-facing changes

- Other (please specify in the release note below)

### Release note

No need for release note.

### CODE REVIEW

**Comment by Xuanwo** [comment]
cc @wcy-fdu, would like to take a review? Thanks in advance.

File: Cargo.lock [hunk 1]
```diff
@@ -708,11 +708,11 @@ dependencies = [
...
  "hyper",
- "hyper-rustls",
+ "hyper-rustls 0.23.2",
  "hyper-tls",
  "lazy_static",
  "pin-project-lite",
- "rustls",
+ "rustls 0.20.8",
  "tokio",
  "tower",
```
Comment (xxchan) [inline Cargo.lock:717]: Is it possible to avoid these duplicated dependencies? 🥲

Comment (Xuanwo) [reply Cargo.lock:717]: Let me take a look.

Comment (Xuanwo) [reply Cargo.lock:717]: `aws-smithy-client` depends on `hyper-rustls 0.23` directly...

https://docs.rs/crate/aws-smithy-client/0.55.3/source/Cargo.toml

Comment (xxchan) [reply Cargo.lock:717]: Get it. Sorry that I misread it. I thought it's `opendal` but it's `aws-smithy-client` 😇

**Review by wcy-fdu — APPROVED** [review]
Sorry for the late review, LGTM!

BTW, we are doing longevity test on gcs and azblob this week, then RisingWave will release 1.0. If you want to bump a major version later, please contact me, and we will do some long running tests to ensure stability. Again, thanks for you PR!

**Comment by Xuanwo** [comment]
> BTW, we are doing longevity test on gcs and azblob this week, then RisingWave will release 1.0. If you want to bump a major version later, please contact me, and we will do some long running tests to ensure stability. Again, thanks for you PR!

Gocha!

**Pull request merged**